tasks/main.yml

```diff
@@ -25,6 +25,14 @@
   set_fact:
     haproxy_version: "{{ '1.5' if '1.5.' in haproxy_version_result.stdout else '1.4' }}"
 
+- name: Copy SSL certificate
+  copy:
+    src: "{{ haproxy_ssl_pem }}"
+    dest: /etc/haproxy/server.pem
+    mode: 0644
+  notify: restart haproxy
+  when: haproxy_ssl_pem != false
```

Comment:
Change `false` to `''` to check against empty string after changing the variable default to an empty string.